cou/steps/analyze.py

```diff
@@ -188,3 +188,38 @@ def _get_minimum_cloud_series(self) -> Optional[str]:
             (app.series for app in self.apps_control_plane + self.apps_data_plane),
             default=None,
         )
+
+    @property
+    def data_plane_machines(self) -> dict[str, juju_utils.Machine]:
+        """Data-plane machines of the model.
+
+        :return: Data-plane machines of the model.
+        :rtype: dict[str, Machine]
+        """
+        return {
+            machine_id: app.machines[machine_id]
+            for app in self.apps_data_plane
+            for machine_id in app.machines
+        }
+
+    @property
+    def control_plane_machines(self) -> dict[str, juju_utils.Machine]:
+        """Control-plane machines of the model.
+
+        :return: Control-plane machines of the model.
+        :rtype: dict[str, Machine]
+        """
+        return {
+            machine_id: app.machines[machine_id]
+            for app in self.apps_control_plane
+            for machine_id in app.machines
+        }
+
+    @property
+    def machines(self) -> dict[str, juju_utils.Machine]:
+        """All machines of the model.
```

Comment:
nit: there could be other machines in the model which don't belong to OpenStack, so maybe we should adjust the wording here to be more accurate:
 ```suggestion
        """All OpenStack machines of the model.
```
also in the `return` docstring